crates/goose/src/agents/recipe_tools/dynamic_task_tools.rs

```diff
@@ -63,18 +63,23 @@ pub struct TaskParameter {
     pub description: Option<String>,
 
     #[serde(skip_serializing_if = "Option::is_none")]
+    #[schemars(with = "Option<Vec<serde_json::Map<String, Value>>>")]
```

Comment:
I think the right fix is to change the `Value`s here to Map<String, Value>, not just in the schemars annotation but in the actual type

Reply:
we can always fix things up in the provider formatter as we do for google, but here we just need to generate the correct schema, not use Value